harper-core/src/parsers/cases.rs

```diff
@@ -0,0 +1,247 @@
+use crate::Lrc;
+use std::collections::VecDeque;
+
+use itertools::Itertools;
+
+use super::{Parser, TokenKind};
+use crate::patterns::{PatternExt, SequencePattern};
+use crate::{Dictionary, FullDictionary, MergedDictionary, Span, Token, VecExt, WordMetadata};
+
+/// A parser that wraps any other parser to collapse token strings that match
+/// the pattern word_word or word-word.
+pub struct Cases {
```

Comment:
Is there a more descriptive name for `Cases`? Maybe something like `CollapseIdentifiers`?